forgerock-auth/src/androidTest/java/org/forgerock/android/auth/callback/DeviceSigningVerifierCallbackTest.java

```diff
@@ -0,0 +1,712 @@
+
+package org.forgerock.android.auth.callback;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import android.content.Context;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.nimbusds.jwt.JWTParser;
+
+import org.assertj.core.api.Assertions;
+import org.forgerock.android.auth.FRAuth;
+import org.forgerock.android.auth.FRListener;
+import org.forgerock.android.auth.FROptions;
+import org.forgerock.android.auth.FROptionsBuilder;
+import org.forgerock.android.auth.FRSession;
+import org.forgerock.android.auth.Logger;
+import org.forgerock.android.auth.Node;
+import org.forgerock.android.auth.NodeListener;
+import org.forgerock.android.auth.NodeListenerFuture;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.util.Base64;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+public class DeviceSigningVerifierCallbackTest {
+    protected Context context = ApplicationProvider.getApplicationContext();
+
+    // This test uses dynamic configuration with the following settings:
+    protected final String AM_URL = "https://openam-dbind.forgeblocks.com/am";
+    protected final String REALM = "alpha";
+    protected final String OAUTH_CLIENT = "AndroidTest";
+    protected final String OAUTH_REDIRECT_URI = "org.forgerock.demo:/oauth2redirect";
+    protected final String SCOPE = "openid profile email address phone";
+    protected final String TREE = "device-verifier";
+
+    protected static String USERNAME = "sdkuser";
+
+    protected static boolean BIND_NEEDED = true; // Used to bind a device only once during the tests execution
+    protected static String KID = null; // Store the kid of the key generated during binding
+
+
+    @Rule
+    public Timeout timeout = new Timeout(10000, TimeUnit.MILLISECONDS);
+
+    @Before
```

Comment:
It is ok to use BIND_NEEDED to control the number of bindings, may consider using BeforeClass annotation, BeforeClass will only run once.

Reply:
Done.